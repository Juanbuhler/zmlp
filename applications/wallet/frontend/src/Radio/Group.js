import PropTypes from 'prop-types'

import { spacing, typography } from '../Styles'

import Radio from '.'

const RadioGroup = ({ legend, options, onClick }) => {
  return (
    <fieldset
      css={{
        display: 'flex',
        alignItems: 'center',
        border: 'none',
        padding: 0,
        margin: 0,
      }}
    >
      <legend
        css={{
          display: 'flex',
          padding: 0,
          paddingRight: spacing.normal,
          float: 'left',
          fontSize: typography.size.medium,
          lineHeight: typography.height.medium,
          fontWeight: typography.weight.medium,
        }}
      >
        {`${legend}:`}
      </legend>

      {options.map((option) => (
        <div key={option.value} css={{ paddingRight: spacing.comfy }}>
          <Radio name={legend} option={option} onClick={onClick} />
        </div>
      ))}
    </fieldset>
  )
}

RadioGroup.propTypes = {
  legend: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
      legend: PropTypes.string.isRequired,
      initialValue: PropTypes.bool.isRequired,
    }).isRequired,
  ).isRequired,
  onClick: PropTypes.func.isRequired,
}

export default RadioGroup
