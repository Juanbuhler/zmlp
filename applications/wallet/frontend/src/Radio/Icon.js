import PropTypes from 'prop-types'

import { colors, constants } from '../Styles'

const RADIO_BUTTON_SIZE = 16
const RADIO_BUTTION_FILL_SIZE = 8

const RadioIcon = ({ value, isChecked }) => {
  return (
    <div
      css={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
      }}
    >
      <input
        type="radio"
        id={value}
        value={value}
        defaultChecked={isChecked}
        css={{
          margin: 0,
          padding: 0,
          WebkitAppearance: 'none',
          borderRadius: RADIO_BUTTON_SIZE,
          width: RADIO_BUTTON_SIZE,
          height: RADIO_BUTTON_SIZE,
          border: constants.borders.radio,
        }}
      />
      <div
        css={{
          position: 'absolute',
          width: RADIO_BUTTION_FILL_SIZE,
          height: RADIO_BUTTION_FILL_SIZE,
          transition: 'all .3s ease',
          opacity: 100,
          borderRadius: RADIO_BUTTON_SIZE,
          backgroundColor: isChecked ? colors.key.one : 'none',
        }}
      />
    </div>
  )
}

RadioIcon.propTypes = {
  value: PropTypes.string.isRequired,
  isChecked: PropTypes.bool.isRequired,
}

export default RadioIcon
