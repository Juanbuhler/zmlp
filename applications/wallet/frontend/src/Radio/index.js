import PropTypes from 'prop-types'

import { colors, spacing, typography } from '../Styles'

import RadioIcon from './Icon'

const Radio = ({ option: { value, label, legend, initialValue }, onClick }) => {
  return (
    <div>
      <label
        css={{ display: 'flex', alignItems: legend ? 'flex-start' : 'center' }}
      >
        <RadioIcon value={value} isChecked={initialValue} onClick={onClick} />

        <div css={{ paddingLeft: spacing.moderate }}>
          <div
            css={{
              fontWeight: legend
                ? typography.weight.bold
                : typography.weight.regular,
              color: colors.structure.white,
            }}
          >
            {label}
          </div>

          <div css={{ color: colors.structure.zinc }}>{legend}</div>
        </div>
      </label>
    </div>
  )
}

Radio.propTypes = {
  option: PropTypes.shape({
    value: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    legend: PropTypes.string.isRequired,
    initialValue: PropTypes.bool.isRequired,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Radio
