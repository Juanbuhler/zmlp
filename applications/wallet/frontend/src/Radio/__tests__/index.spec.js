import TestRenderer, { act } from 'react-test-renderer'

import Radio from '..'

describe('<Radio />', () => {
  it('should render properly', () => {
    const mockFn = jest.fn()

    const component = TestRenderer.create(
      <Radio
        option={{
          label: 'Radio',
          value: 'radio',
          legend: 'radioLegend',
          initialValue: false,
          isDisabled: false,
        }}
        onClick={mockFn}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root.findByProps({ type: 'radio' }).props.onClick()
    })

    expect(component.toJSON()).toMatchSnapshot()

    expect(mockFn).toHaveBeenCalledWith(true)
  })

  it('should render properly disabled', () => {
    const mockFn = jest.fn()

    const component = TestRenderer.create(
      <Radio
        option={{
          label: 'Radio',
          value: 'radio',
          legend: 'radioLegend',
          initialValue: false,
          isDisabled: true,
        }}
        onClick={mockFn}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root.findByProps({ type: 'radio' }).props.onClick()
    })

    expect(mockFn).not.toHaveBeenCalled()
  })

  it('should render properly checked and disabled', () => {
    const mockFn = jest.fn()

    const component = TestRenderer.create(
      <Radio
        option={{
          label: 'Radio',
          value: 'radio',
          legend: 'radioLegend',
          initialValue: true,
          isDisabled: true,
        }}
        onClick={mockFn}
      />,
    )

    expect(component.toJSON()).toMatchSnapshot()

    act(() => {
      component.root.findByProps({ type: 'radio' }).props.onClick()
    })

    expect(mockFn).not.toHaveBeenCalled()
  })
})
