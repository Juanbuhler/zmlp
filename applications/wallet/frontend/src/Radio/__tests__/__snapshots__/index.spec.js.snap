// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should render properly checked 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  padding-top: 1px;
}

.emotion-2 {
  padding: 0;
  -webkit-appearance: none;
  border-radius: 16px;
  width: 16px;
  height: 16px;
  border: 1px solid #ffffff;
}

.emotion-3 {
  position: absolute;
  width: 8px;
  height: 8px;
  opacity: 100;
  border-radius: 16px;
  background-color: #3A5DFF;
}

.emotion-4 {
  padding-left: 12px;
}

.emotion-5 {
  font-weight: 700;
  color: #ffffff;
}

.emotion-6 {
  color: #b3b3b3;
}

<div>
  <label
    className="emotion-0"
  >
    <div
      className="emotion-1"
    >
      <input
        className="emotion-2"
        defaultChecked={true}
        name="radio"
        onClick={[Function]}
        type="radio"
        value="radio"
      />
      <div
        className="emotion-3"
      />
    </div>
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        Radio
      </div>
      <div
        className="emotion-6"
      >
        radioLegend
      </div>
    </div>
  </label>
</div>
`;

exports[`<Radio /> should render properly unchecked 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  padding-top: 1px;
}

.emotion-2 {
  padding: 0;
  -webkit-appearance: none;
  border-radius: 16px;
  width: 16px;
  height: 16px;
  border: 1px solid #808080;
}

.emotion-3 {
  position: absolute;
  width: 8px;
  height: 8px;
  opacity: 100;
  border-radius: 16px;
  background-color: none;
}

.emotion-4 {
  padding-left: 12px;
}

.emotion-5 {
  font-weight: 700;
  color: #ffffff;
}

.emotion-6 {
  color: #b3b3b3;
}

<div>
  <label
    className="emotion-0"
  >
    <div
      className="emotion-1"
    >
      <input
        className="emotion-2"
        defaultChecked={false}
        name="radio"
        onClick={[Function]}
        type="radio"
        value="radio"
      />
      <div
        className="emotion-3"
      />
    </div>
    <div
      className="emotion-4"
    >
      <div
        className="emotion-5"
      >
        Radio
      </div>
      <div
        className="emotion-6"
      >
        radioLegend
      </div>
    </div>
  </label>
</div>
`;
